docs/proposals/scheduling/20220510-load-aware-scheduling.md

```diff
@@ -0,0 +1,137 @@
+
+Here is an explanation of why the allocated Pods are also involved in the estimation in step 4. After the Pod is scheduled, it needs to pull the image and mount the disk before it starts. After many seconds have passed since the Pod serving, it will cause the koordlet to report NodeMetrics without missing usages of these Pods. At this time if the scheduler continues to schedule Pods on these nodes, it is likely that the future resource utilization will exceed the threshold. In order to avoid this situation, the plugin implements the Reserve extension point to records the allocated time of the Pod, denoted as *R*, and denote the latest update time of NodeMetric as *U*, then when scheduling the next Pod, the allocated Pods in the current node need to participate in the estimation which meet the conditions *(R > U or U - R > nodeMetricUpdateWindow)*.
+
+If the last update time of the current node's NodeMetric exceeds the configured maximum update time window, the node will get 0 points.
+
+#### Plugin configuration
+
+```go
+
+type LoadAwareSchedulingArgs struct {
+  metav1.TypeMeta
+
+  FilterUnhealthyNodeMetrics         bool                          `json:"filterUnhealthyNodeMetrics,omitempty"`
+  NodeMetricUpdateWindowInSeconds    int64                         `json:"nodeMetricUpdateWindowInSeconds,omitempty"`
+  NodeMetricUpdateMaxWindowInSeconds int64                         `json:"nodeMetricUpdateMaxWindowInSeconds,omitempty"`
+  ResourceWeights                    map[corev1.ResourceName]int64 `json:"resourceWeights,omitempty"`
+  UsageThresholds                    map[corev1.ResourceName]int64 `json:"usageThresholds,omitempty"`
+  EstimatedScalingFactors            map[corev1.ResourceName]int64 `json:"estimatedScalingFactors,omitempty"`
+}
+
+```
+
+- `FilterUnhealthyNodeMetrics` indicates whether to filter nodes where koordlet fails to update NodeMetric.
+- `NodeMetricUpdateWindowInSeconds` indicates the period that koordlet updates NodeMetric, the default is 60s.
+- `NodeMetricUpdateMaxWindowInSeconds` indicates the tolerable maximum period for koordlet to update NodeMetric. If it exceeds, the node is considered abnormal.
+- `ResourceWeights` indicates the weights of resources. The weights of CPU and Memory are both 1 by default.
+- `UsageThresholds` indicates the resource utilization threshold, the default for CPU is 65%, and the default for memory is 95%.
+- `EstimatedScalingFactors` indicates the factor when estimating resource usage. The default value of CPU is 85%, and the default value of Memory is 70%.
+
+#### Custom usage thresholds
+Definition Annotation supports user-defined resource utilization thresholds for nodes, and responds to scenarios due to differences in models, or grayscale, and testing.
+
+```go
+const (
+  AnnotationCustomUsageThresholds = "scheduling.koordinator.sh/usage-thresholds"
+)
+
+type CustomUsageThresholds struct {
+  UsageThresholds map[corev1.ResourceName]int64 `json:"usageThresholds,omitempty"`
+}
+```
```

Comment:
Add a section to describe the consideration of node resource profiling, such as whether to use real-time utilization or 3-sigma profiling.